lib/OpenQA/Script/CloneJobSUSE.pm

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Script::CloneJobSUSE;
+use Mojo::Base -strict, -signatures;
+use Data::Dump 'pp';
+use Exporter 'import';
+
+our @EXPORT = qw(
+  detect_maintenance_update
+);
+
+sub collect_incident_repos ($url_handler, $settings) {
+    my @incident_urls;
+    my $ua = $url_handler->{ua};
+    for my $setting (keys %$settings) {
+        if ($setting =~ /(.*)TEST_TEMPLATE/) {
+            next unless $settings->{$1 . 'TEST_ISSUES'};
+            foreach my $incident (split(/,/, $settings->{$1 . 'TEST_ISSUES'})) {
+                my $test_template_url = $settings->{$setting};
+                $test_template_url =~ s/\@INCIDENTNR\@/$incident/;
+                my $response = $ua->get($test_template_url);
+                if (!$response->is_success) {
+                    push @incident_urls, $test_template_url;
+                }
+            }
+        }
+
+    }
+    return \@incident_urls;
+}
+
+sub detect_maintenance_update ($jobid, $url_handler, $settings) {
+    my $urls = collect_incident_repos($url_handler, $settings);
+    if (@$urls) {
+        print STDERR "Current job $jobid will fail, because the below updates have already been released\n", pp($urls);
+        return 0;
+    }
+    return 1;
```

Comment:
How about

```suggestion
    die "Current job $jobid will fail, because the below updates have already been released\n" . pp($urls) if @$urls;
```

Reply:
Seconded. You are not doing anything with those return codes; better go with `die` as @okurz is suggesting and you also avoid `print STDERR`.

Reply:
Well, maybe we need to collect experiences from real-live operations to know for sure. I am ok to leave as is for now.